### PR TITLE
feat: Add Cava spider

### LINKED ISSUE #3515: Cava
https://cava.com/locations

### PR DESCRIPTION
**Pull request by rubyruins**
Fixes #3515.

### CODE REVIEW

**Comment by rubyruins** [comment]
@iandees, could you please review this?

File: locations/spiders/cava.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                opening_hours = opening_hours.split("//")
+            opening_hours = self.parse_opening_hours(opening_hours)
+
+        properties = {
+            "ref": name,
+            "name": name,
+            "street_address": street_address,
+            "postcode": postcode,
+            "state": state,
+            "addr_full": addr_full,
```
Comment (iandees) [inline locations/spiders/cava.py:134]: We don't need `addr_full` if we have `street_address`, `city`, and `state`.

File: locations/spiders/cava.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            .replace(nonBreakingSpace, " ")
+            + ", "
+            + location.xpath('.//span[@class="locality"]/text()')
+            .extract_first()
+            .replace(nonBreakingSpace, " ")
+        )
+        state = STATES.get(
+            location.xpath('.//span[@class="region"]/text()').extract_first().strip(),
+            "",
+        )
```
Comment (iandees) [inline locations/spiders/cava.py:108]: I appreciate the work you did to add the `STATES` mapping, but we don't need to expand the two-letter code for the state to a full name.

File: locations/spiders/cava.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    def parse_state(self, state):
+        location_selectors = state.xpath('.//div[@class="vcard"]')
+        for location_selector in location_selectors:
+            yield from self.parse_location(location_selector)
+
+    def parse_location(self, location):
+        nonBreakingSpace = "\xa0"
+        name = location.xpath(".//h3/text()").extract_first()
+        street_address = (
+            location.xpath('.//div[@class="street-address"]/text()')
```
Comment (iandees) [inline locations/spiders/cava.py:97]: `street_address` should only be the `street-address` data and not include the locality.

File: locations/spiders/cava.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            .extract_first()
+            .replace(nonBreakingSpace, " ")
+            + ", "
+            + location.xpath('.//span[@class="locality"]/text()')
```
Comment (iandees) [inline locations/spiders/cava.py:101]: `locality` should probably go in `city`.

Comment (rubyruins) [reply locations/spiders/cava.py:101]: Done!

File: locations/spiders/cava.py (hunk 3 above)
Comment (rubyruins) [reply locations/spiders/cava.py:97]: @iandees, the street address currently does not include any specific locality details.

File: locations/spiders/cava.py (hunk 2 above)
Comment (rubyruins) [reply locations/spiders/cava.py:108]: Sure thing. The states are now not expanded into a full name.

File: locations/spiders/cava.py (hunk 1 above)
Comment (rubyruins) [reply locations/spiders/cava.py:134]: Sounds, good, I have removed it.

**Pull request merged**